go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -255,6 +260,27 @@ func (c *Controller) Run(workers int, stopCh <-chan struct{}, wg *sync.WaitGroup
 	return nil
 }
 
+func (c *Controller) Cleanup() {
+	klog.Infof("Shutting down controller %s for network=%s", controllerName, c.netInfo.GetNetworkName())
+	c.queue.ShutDown()
+
+	if c.nodeHandler != nil {
```

Comment:
Were you able to determine if other UDN controllers suffer from this?
I think https://github.com/ovn-kubernetes/ovn-kubernetes/blob/9ea5b14aa3b2dcf2e6ebf36bbf3330337cf262e1/go-controller/pkg/ovn/controller/udnenabledsvc/udn_enabled_svc.go#L84 might be the same?

Reply:
good point, I was lazy :)

Reply:
ok, I checked that all controllers that use `AddEventHandler` are only stopped with the whole binary, e.g. this one is only created once with the controller_managaer here https://github.com/ovn-kubernetes/ovn-kubernetes/blob/256b38cc4bd347b5f8ce652c07195f868df614ce/go-controller/pkg/controllermanager/controller_manager.go#L468-L476
even though a lot of controllers using this ^ directly could be refactored, I am not doing it in this pr :)